### ISSUE TITLE
Convert to Java, use Worker API

### ISSUE THREAD

**Issue by benjamin-bader** [KeepSafe/dexcount-gradle-plugin#379]
This is the saddest issue.

Gradle bundles its own Kotlin runtime and makes no guarantee about compatibility with third-party Kotlin code.  Per Gradle, the _only_ safe and forwards-compatible way to use Kotlin in a plugin is via the Worker API, using classpath isolation.

This means that our code _cannot_ be Kotlin, and since we have Kotlin dependencies, they can only safely be used from behind that classpath-isolated worker.

The plan:
- Convert `Plugin.kt` to several Java classes - a base Plugin, and moving the applicators to individual Java files
  - the official AGP API is Kotlin which is going to be trouble.  Java's the lowest common denominator, so we can do it, but it'll be ugly
- Either convert the tasks to Java, or move them to an impl module and depend on it dynamically.  Probably better to just move them over to Java.

Checklist:

- [x] DexCountExtension
- [x] DexCountException
- [x] Deobfuscator
- [x] ColorConsole.kt
- [x] GradleApi.kt
- [x] OutputFormat
- [x] PackageTree
- [x] PrintOptions
- [x] SourceFile.kt
- [x] Thrifty.kt (maybe we can solve this by bumping to 3.0.0-RC01 which has this stuff built in?)
- [x] GeneratePackageTreeTask
- [x] DexCountOutputTask
- [x] CountReporter
- [x] DexMethodCountPlugin
- [x] TaskApplicators

**Comment by nevack** [comment]
Seems quite painful

**Comment by benjamin-bader** [comment]
Yep.  Very much so.  This is the second major language change for this project.  Initially it was Groovy, because who wants to write Java?  Eventually, I got so fed up with maintaining hacks around Gradle's Groovy version changes that I jumped to Kotlin as soon as possible.  I have no desire to repeat the hacks a second time around.

When a principal engineer at Gradle tells me "Kotlin runtimes are controlled by us now and are subject to change", I see the writing on the wall.
